src/coreclr/gc/gc.cpp

```diff
@@ -8368,29 +8391,262 @@ void destroy_card_table (uint32_t* c_table)
     dprintf (2, ("Table Virtual Free : %Ix", (size_t)&card_table_refcount(c_table)));
 }
 
-uint32_t* gc_heap::make_card_table (uint8_t* start, uint8_t* end)
+void gc_heap::get_card_table_element_sizes (uint8_t* start, uint8_t* end, size_t sizes[total_bookkeeping_elements])
 {
-    assert (g_gc_lowest_address == start);
-    assert (g_gc_highest_address == end);
+    sizes[card_table_element] = size_card_of (start, end);
+    sizes[brick_table_element] = size_brick_of (start, end);
 
-    uint32_t virtual_reserve_flags = VirtualReserveFlags::None;
+    sizes[brick_table_element + 1] = 0;
```

Comment:
you could memset on sizes to begin with so you don't need to set certain fields to 0.